packages/typespec-vscode/src/telemetry/telemetry-client.ts

```diff
@@ -130,40 +155,57 @@ class TelemetryClient {
     }
   }
 
-  public logOperationTelemetryEvent(event: OperationTelemetryEvent) {
-    const telFunc =
-      event.result === "success" || event.result === "cancelled"
-        ? this.sendEvent
-        : this.sendErrorEvent;
-    telFunc.call(this, event.eventName, {
-      activityId: isWhitespaceStringOrUndefined(event.activityId)
-        ? emptyActivityId
-        : event.activityId!,
-      // ISO format: YYYY-MM-DDTHH:mm:ss.sssZ
-      startTime: event.startTime.toISOString(),
-      endTime: event.endTime?.toISOString() ?? "",
-      lastStep: event.lastStep ?? "undefined",
-      result: event.result ?? "undefined",
-    });
+  public logOperationTelemetryEvent(event: OperationTelemetryEvent, delay: boolean = false) {
+    const raw: RawTelemetryEvent = {
+      eventName: event.eventName,
+      properties: {
+        activityId: isWhitespaceStringOrUndefined(event.activityId)
+          ? emptyActivityId
+          : event.activityId!,
+        // ISO format: YYYY-MM-DDTHH:mm:ss.sssZ
+        startTime: event.startTime.toISOString(),
+        endTime: event.endTime?.toISOString() ?? "",
+        lastStep: event.lastStep ?? "undefined",
```

Comment:
for the next 2 do you need ?? undefined

Reply:
put them as "undefined" string explicitly to make it more clear in the final data in telemetry store. Otherwise, these properties will be omitted in the store which is confusing sometimes.

Reply:
i see, didn't noticed undefined was a string. Does feel weird to do that as well I guess but i don't fully get where it gets exposed later so if you think its better this way.